opensquirrel/exporter/quantify_scheduler_exporter.py

```diff
@@ -86,21 +82,30 @@ def visit_controlled_gate(self, g: ControlledGate) -> None:
             )
             return
 
-        raise _unsupported_gates_exception
+        raise UnsupportedGateError(
+            f"Cannot export circuit: it contains unsupported gates. The following gate could not be exported: {g}. "
+            "Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
+        )
 
 
 def export(circuit: Circuit) -> "quantify_scheduler.Schedule":
     if "quantify_scheduler" not in globals():
 
         class QuantifySchedulerNotInstalled:
             def __getattr__(self, attr_name: Any) -> None:
-                raise ImportError("quantify-scheduler is not installed, or cannot be installed on your system")
+                raise ModuleNotFoundError("quantify-scheduler is not installed, or cannot be installed on your system")
 
         global quantify_scheduler
         quantify_scheduler = QuantifySchedulerNotInstalled()
         global quantify_scheduler_gates
         quantify_scheduler_gates = QuantifySchedulerNotInstalled()
 
     schedule_creator = _ScheduleCreator(circuit.qubit_register_name)
-    circuit.ir.accept(schedule_creator)
+    try:
+        circuit.ir.accept(schedule_creator)
+    except UnsupportedGateError as e:
+        raise ExportError(
+            "Cannot export circuit: it contains unsupported gates. The following gate could not be exported: "
+            f"{e.unsupported_gate}. Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
```

Comment:
```suggestion
            f"cannot export circuit: {e}. Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"
```

I would just print the error message:
- The `it contains unsupported gates` text is redundant.
- The `The following gate could not be exported: ` should como from `UnsupportedGateError` class.
- All in all, we could read: `cannot export circuit: the following gate is not supported: blah. Decompose...`
- This way, we could also reuse the same message for different types of exceptions. In case that couldn't be possible (e.g., the `Decompose all gates...` part could only make sense for UnsupportedGateError exceptions), then we could create a QuantifySchedulerUnsupportedGateError that would add that last piece of advise.

Note also that error messages should start in lowercase, because they will usually be suffixed (with `Error: `, or `Cannot export circuit: `, etc.)

Reply:
I changed the error message to be a bit more concise. However, I do think that `Decompose all gates to the Quantify-scheduler gate set first (rxy, rz, cnot, cz)"` is relevant here, because we only raise the error with this specific message when we catch an `UnsupportedGateError`. I would really prefer to keep the Error types to general concepts, and the messages really precise.

Reply:
Yes, yes, I wouldn't remove the `Decompose all gates...` part. Thanks!